sample_metadata/parser/generic_parser.py

```diff
@@ -73,7 +73,7 @@
 
 # construct rmatch string to capture all fastq patterns
 rmatch_str = (
-    r'[_\.-][Rr]?[12]('
+    r'[_\.-][Rr]?[12](_\d+)?('
```

Comment:
Can you add some examples in the doctest down here with the new format?

https://github.com/populationgenomics/sample-metadata/blob/299bc966b16e9f57db80f0f8cbafe2635b721d6c/sample_metadata/parser/generic_parser.py#L1145

Reply:
I suspect this won't correctly group on the second group (the 001 after the R\d)

Reply:
@illusional not sure how to see if the doctest passed here but I've added an example with _R1/R2_001.fastq.gz, is this sufficient? Cheers

Reply:
That format is typically used to represent a sharded fastq - ie we want all files with the prefix `21R2112345-20210326-A00123_S70_L001_R1_*` to be merged in the final alignment.

Reply:
Put another way, they should end up in the same read group in the bam.

Reply:
I have been practicing with the fastq parser to try and get this to work. So the current function creates pairs of R1 / R2 files based on their common filename prefix

With this change, do we want to create pairs of reads in both directions with a common prefix and a common suffix? 
i.e group `R1_001` with `R2_001` etc.
```
[
['L001_R1_001.fastq.gz', 'L001_R2_001.fastq.gz'],  
['L001_R1_002.fastq.gz', 'L001_R2_002.fastq.gz']
]
```

Or should the pairs of reads be the same direction but with different suffix?
i.e group `R1_001` with `R1_002`
```
[
['L001_R1_001.fastq.gz', 'L001_R1_002.fastq.gz'],  
['L001_R2_001.fastq.gz', 'L001_R2_002.fastq.gz']
]
```

Or is it neither of the above and i'm misunderstanding?

Reply:
As far as I know option 1 is correct, these are just normal fastq pairs. Unless I am missing something @illusional?